### PR TITLE
refactor(core)!: `tuiFormatNumber(value, configs)` new function signature

### PR DESCRIPTION
**Pull request by nsbarsukov**
## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] The commit message follows [Conventional Commits](https://www.conventionalcommits.org/en/)
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type

What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [X] Refactoring
- [ ] Code style update
- [ ] Build or CI related changes
- [ ] Documentation content changes

## What is the current behavior?
**Utility:**
```ts
function tuiFormatNumber(
    value: number,
    decimalLimit: number | null = null,
    decimalSeparator: string = `,`,
    thousandSeparator: string = CHAR_NO_BREAK_SPACE,
    zeroPadding: boolean = true,
): string;
```

**Pipe:**
```html
<p>Formatted number: {{ 10500.33 | tuiFormatNumber: 4:'.' }}</p>
```

## What is the new behavior?
**Utility:**
```ts
function tuiFormatNumber(
    value: number,
    {
        decimalLimit = null,
        decimalSeparator = `,`,
        thousandSeparator = CHAR_NO_BREAK_SPACE,
        zeroPadding = true,
    }: Partial<TuiNumberFormatSettings> = {},
): string
```

**Pipe:**
```html
<p>Formatted number: {{ 10500.33 | tuiFormatNumber: {decimalLimit: 4, decimalSeparator: '.'} }}</p>
```

## Does this PR introduce a breaking change?

- [X] Yes
- [ ] No

### CODE REVIEW

**Review by splincode — APPROVED** [review]

File: projects/core/utils/format/format-number.ts [hunk 1]
```diff
@@ -1,25 +1,27 @@
...
-    decimalLimit: number | null = null,
-    decimalSeparator: string = `,`,
-    thousandSeparator: string = CHAR_NO_BREAK_SPACE,
-    zeroPadding: boolean = true,
+    {
+        decimalLimit = null,
+        decimalSeparator = `,`,
+        thousandSeparator = CHAR_NO_BREAK_SPACE,
+        zeroPadding = true,
+    }: Partial<TuiNumberFormatSettings & {decimalLimit: number | null}> = {},
```
Comment (nsbarsukov) [inline projects/core/utils/format/format-number.ts:24]: Move `decimalLimit: number | null` to `TuiNumberFormatSettings`-interface too

File: projects/core/pipes/format-number/format-number.pipe.ts [hunk 2]
```diff
@@ -14,24 +14,26 @@ export class TuiFormatNumberPipe implements PipeTransform {
...
         value: number,
         decimalLimit: number | null = null,
-        decimalSeparator: string = this.numberFormat.decimalSeparator,
-        thousandSeparator: string = this.numberFormat.thousandSeparator,
-        zeroPadding: boolean = this.numberFormat.zeroPadding,
+        decimalSeparator: TuiNumberFormatSettings['decimalSeparator'] = this.numberFormat
+            .decimalSeparator,
+        thousandSeparator: TuiNumberFormatSettings['thousandSeparator'] = this
+            .numberFormat.thousandSeparator,
+        zeroPadding: TuiNumberFormatSettings['zeroPadding'] = this.numberFormat
```
Comment (nsbarsukov) [inline projects/core/pipes/format-number/format-number.pipe.ts:29]: replace all optional arguments with single object (with all configurations)

File: projects/core/utils/format/format-number.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/core/utils/format/format-number.ts:24]: Replace 
```ts
// number of digits of decimal part,
// null to keep untouched
decimalLimit: number | null = null
```
with
```ts
// number of digits of decimal part,
// Infinity to keep untouched
decimalLimit: number = Infinity
```

**Review by vladimirpotekhin — APPROVED** [review]

File: projects/core/utils/format/format-number.ts (hunk 1 above)
Comment (nsbarsukov) [reply projects/core/utils/format/format-number.ts]: ✅

File: projects/core/pipes/format-number/format-number.pipe.ts (hunk 2 above)
Comment (nsbarsukov) [reply projects/core/pipes/format-number/format-number.pipe.ts]: done✅

**Review by splincode — APPROVED** [review]

**Review by waterplea — APPROVED** [review]

**Pull request merged**